src/gui/properties/propertieswidget.cpp

```diff
@@ -173,6 +175,39 @@ PropertiesWidget::~PropertiesWidget()
     delete m_ui;
 }
 
+void PropertiesWidget::displayToggleColumnsMenu(const QPoint &)
+{
+    QMenu *menu = new QMenu(this);
+    menu->setAttribute(Qt::WA_DeleteOnClose);
+    menu->setTitle(tr("Column visibility"));
+
+    for (int i = 0; i < TorrentContentModelItem::TreeItemColumns::NB_COL; ++i)
+    {
+        QAction *myAct = menu->addAction(m_propListModel->headerData(i, Qt::Horizontal, Qt::DisplayRole).toString());
+        myAct->setCheckable(true);
+
+        if (i == TorrentContentModelItem::TreeItemColumns::COL_NAME)
+            myAct->setEnabled(false);
+
+        myAct->setChecked(!m_ui->filesList->isColumnHidden(i));
+        myAct->setData(i);
+    }
+
+    connect(menu, &QMenu::triggered, this, [this](const QAction *action)
+    {
+        const int col = action->data().toInt();
+
+        m_ui->filesList->setColumnHidden(col, !m_ui->filesList->isColumnHidden(col));
+
+        if (!m_ui->filesList->isColumnHidden(col) && (m_ui->filesList->columnWidth(col) <= 5))
+            m_ui->filesList->resizeColumnToContents(col);
+
+        saveSettings();
```

Comment:
Does it affect file list columns visibility?

Reply:
Yes, `saveSettings` saves the splitter position, the file list headers [position\size\visibility](https://github.com/qbittorrent/qBittorrent/blob/9bfc74a1bcf1420ea0393ffbafe5057cae4f37e4/src/gui/properties/propertieswidget.cpp#L387) and the current selected tab.